app/src/main/java/com/example/kotlin/movieapp/adapter/BaseAdapter.kt

```diff
@@ -0,0 +1,46 @@
+package com.example.kotlin.movieapp.adapter
+
+import android.databinding.DataBindingUtil
+import android.databinding.ObservableList
+import android.databinding.ViewDataBinding
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import com.example.kotlin.movieapp.BR
+
+abstract class BaseAdapter<T>(val items: ObservableList<T>): RecyclerView.Adapter<BaseViewHolder>() {
+
+    abstract fun getLayout(): Int
+
+    abstract fun getItemForPosition(position: Int): T
+
+    val onListChangeListener = object: ObservableList.OnListChangedCallback<ObservableList<T>>() {
+        override fun onItemRangeRemoved(sender: ObservableList<T>?, positionStart: Int, itemCount: Int) = notifyItemRangeChanged(positionStart, itemCount)
+
```

Comment:
you can probably remove all spaces between each of these methods to make this more compact:
```
override fun onItemRangeRemoved()...
override fun onChanged()...
...
```